### PR TITLE
[Feature] Adding Vehicle Pose estimation using CARFUSION Dataset

### PR DESCRIPTION
**Pull request by dineshreddy91**
## Checklist

**Before PR**:

- [x] I have read and followed the workflow indicated in the [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) to create this PR.
- [x] Pre-commit or linting tools indicated in [CONTRIBUTING.md](https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md) are used to fix the potential lint issues.
- [x] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit tests to ensure correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [x] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

**Comment by jin-s13** [comment]
Thank you very much for your contribution! We will review the codes ASAP. 
It seems that the lint check fails. Could you please follow https://github.com/open-mmlab/mmpose/blob/master/.github/CONTRIBUTING.md this guide to use pre-commit to format your codes?

**Comment by jin-s13** [comment]
Also, we would appreciate it, if you could add some basic unittest.
You may refer to https://github.com/open-mmlab/mmpose/blob/master/tests/test_datasets/test_top_down_dataset.py
Thanks!

**Comment by jin-s13** [comment]
Have you already trained the HRNet model using MMPose? 
Could you upload the trained models and logs to google drive or onedrive? This will be very helpful for our accuracy verification.

**Comment by dineshreddy91** [comment]
> 

The data and the config file can be found here: https://drive.google.com/drive/folders/1ikzZ6QXVS5U0kruBX843tMlKpCDWDRU5?usp=sharing

File: mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+        0: 'Right Front wheel',
+        1: 'Left Front wheel'
+        2: 'Right Back wheel'
+        3: 'Left Back wheel'
+        4: 'Right Front HeadLight'
+        5: 'Left Front HeadLight'
+        6: 'Right Back HeadLight'
+        7: 'Left Back HeadLight'
+        8: 'Exhaust'
+        9:
```
Comment (jin-s13) [inline mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py:34]: Why this line is left blank?

File: mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+from xtcocotools.cocoeval import COCOeval
+
+from ....core.post_processing import oks_nms, soft_oks_nms
+from ...builder import DATASETS
+from ..base import Kpt2dSviewRgbImgTopDownDataset
+
+
+@DATASETS.register_module()
+class TopDownCarFusionDataset(Kpt2dSviewRgbImgTopDownDataset):
+    """CocoDataset dataset for top-down pose estimation.
```
Comment (jin-s13) [inline mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py:19]: CocoDataset --> CarFusion

File: configs/_base_/datasets/carfusion.py [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+            link=('Left Back Top', 'Right Back Top'),
+            id=17,
+            color=[255, 128, 0])
+    },
+    joint_weights=[
+        1., 1., 1., 1., 1., 1., 1., 1.2, 1.2, 1.5, 1.5, 1., 1., 1.2, 1.2, 1.5,
+        1.5
+    ],
+    sigmas=[
+        0.026, 0.025, 0.025, 0.035, 0.035, 0.079, 0.079, 0.072, 0.072, 0.062,
```
Comment (jin-s13) [inline configs/_base_/datasets/carfusion.py:205]: It seems that these sigmas are for COCO dataset. 
Did Carfusion dataset provide official sigma values?

File: configs/_base_/datasets/carfusion.py [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+            id=16,
+            color=[255, 128, 0]),
+        17:
+        dict(
+            link=('Left Back Top', 'Right Back Top'),
+            id=17,
+            color=[255, 128, 0])
+    },
+    joint_weights=[
+        1., 1., 1., 1., 1., 1., 1., 1.2, 1.2, 1.5, 1.5, 1., 1., 1.2, 1.2, 1.5,
```
Comment (jin-s13) [inline configs/_base_/datasets/carfusion.py:201]: I will recommend using all ones for joint_weights.

File: mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py [hunk 5]
```diff
@@ -0,0 +1,403 @@
...
+                'dataset': self.dataset_name,
+                'bbox_score': 1,
+                'bbox_id': bbox_id
+            })
+            bbox_id = bbox_id + 1
+
+        return rec
+
+    def _load_carfusion_vehicle_detection_results(self):
+        """Load coco person detection results."""
```
Comment (jin-s13) [inline mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py:173]: coco person --> carfusion vehicle

File: mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py (hunk 1 above)
Comment (dineshreddy91) [reply mmpose/datasets/datasets/top_down/topdown_carfusion_dataset.py:34]: 9: keypoint is miscellaneous in the dataset

**Comment by dineshreddy91** [comment]
Fixed the  indentation erros and the comments and uploaded the test files. Will update unittest later.

File: configs/_base_/datasets/carfusion.py (hunk 3 above)
Comment (dineshreddy91) [reply configs/_base_/datasets/carfusion.py]: i have changed them to be the same. They are not provided by the dataset

**Comment by dineshreddy91** [comment]
update-model-index.......................................................Failed
- hook id: update-model-index
- exit code: 1
- files were modified by this hook


When updating the file with accuracy number I face this issue during git commit. Code corresponding to this is given below.
Results on Carfusion test dataset

| Arch                                          | Input Size |  AP   | AP<sup>50</sup> | AP<sup>75</sup> |  AR   | AR<sup>50</sup> |                     ckpt                      |                      log                      |
| :-------------------------------------------- | :--------: | :---: | :-------------: | :-------------: | :---: | :-------------: | :-------------------------------------------: | :-------------------------------------------: |
| [pose_hrnet_w32](/configs/vehicle/2d_kpt_sview_rgb_img/topdown_heatmap/carfusion/hrnet_w32_carfusion_384x288.py) |  384x288   | 0.493 |      0.915      |      0.489      | 0.548 |      0.925      | [ckpt]() | [log]() |

let me know how to fix this.

**Comment by jin-s13** [comment]
You may run `pre-commit run --all-files` locally before commit to fix lint errors.

File: configs/vehicle/2d_kpt_sview_rgb_img/topdown_heatmap/carfusion/hrnet_carfusion.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+month = {June},
+year = {2018}
+}
+```
+
+</details>
+
+| Arch                                          | Input Size |  AP   | AP<sup>50</sup> | AP<sup>75</sup> |  AR   | AR<sup>50</sup> |                     ckpt                      |                      log                      |
+| :-------------------------------------------- | :--------: | :---: | :-------------: | :-------------: | :---: | :-------------: | :-------------------------------------------: | :-------------------------------------------: |
+| [pose_hrnet_w32](/configs/vehicle/2d_kpt_sview_rgb_img/topdown_heatmap/carfusion/hrnet_w32_carfusion_384x288.py) |  384x288   | 0.760 |      0.906      |      0.829      | 0.810 |      0.943      | [ckpt](https://download.openmmlab.com/mmpose/top_down/hrnet/hrnet_w32_carfusion_384x288-d9f0d786_20200708.pth) | [log](https://download.openmmlab.com/mmpose/top_down/hrnet/hrnet_w32_carfusion_384x288_20200708.log.json) |
```
Comment (jin-s13) [inline configs/vehicle/2d_kpt_sview_rgb_img/topdown_heatmap/carfusion/hrnet_carfusion.md:37]: How do you obtain this result? Do you use gt bbox during testing？

**Comment by jin-s13** [comment]
Hi @dineshreddy91,
I noticed that you updated the codes a few days ago. 
If you are still interested, let's work together to fix the remaining issues and merge this PR in Nov.

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @dineshreddy91 ！We are grateful for your efforts in helping improve mmpose open-source project during your personal time. 

Welcome to join OpenMMLab Special Interest Group (SIG) private channel on Discord, where you can share your experiences, ideas, and build connections with like-minded peers. To join the SIG channel, simply message moderator— OpenMMLab on Discord or briefly share your open-source contributions in the #introductions channel and we will assist you. Look forward to seeing you there! Join us ：https://discord.gg/UjgXkPWNqA
If you have a WeChat account，welcome to join our community on WeChat. You can add our assistant ：openmmlabwx. Please add "mmsig + Github ID" as a remark when adding friends：）

Thank you again for your contribution❤

**Pull request closed without merging**